corehq/apps/linked_domain/ucr.py

```diff
@@ -10,31 +10,64 @@
 from corehq.apps.linked_domain.exceptions import (
     DomainLinkError,
     MultipleDownstreamAppsError,
+    RegistryNotAccessible,
 )
 from corehq.apps.linked_domain.remote_accessors import \
     get_ucr_config as remote_get_ucr_config
 from corehq.apps.userreports.dbaccessors import (
     get_datasources_for_domain,
+    get_registry_report_configs_for_domain,
     get_report_configs_for_domain,
 )
 from corehq.apps.userreports.models import (
     DataSourceConfiguration,
     ReportConfiguration,
+    is_data_registry_report,
 )
 from corehq.apps.userreports.tasks import rebuild_indicators
 
-LinkedUCRInfo = namedtuple("LinkedUCRInfo", "datasource report")
+from corehq.apps.registry.models import DataRegistry
+
+from corehq.util.couch import DocumentNotFound
+
+from tastypie.exceptions import NotFound
+
+from corehq.apps.userreports.exceptions import (
+    DataSourceConfigurationNotFoundError,
+)
+
+from corehq.apps.userreports.util import (
+    get_report_config_or_not_found,
+    wrap_report_config_by_type,
+    _wrap_data_source_by_doc_type
+)
 
+LinkedUCRInfo = namedtuple("LinkedUCRInfo", "datasource report")
 
 def create_linked_ucr(domain_link, report_config_id):
     if domain_link.is_remote:
         remote_configs = remote_get_ucr_config(domain_link, report_config_id)
         datasource = remote_configs["datasource"]
         report_config = remote_configs["report"]
     else:
-        report_config = ReportConfiguration.get(report_config_id)
-        datasource = DataSourceConfiguration.get(report_config.config_id)
-
+        try:
+            report_config = get_report_config_or_not_found(domain_link.master_domain, report_config_id)
+        except DocumentNotFound:
+            raise NotFound
+        try:
+            datasource = DataSourceConfiguration.get(report_config.config_id)
```

Comment:
shouldn't this use `get_ucr_datasource_config_by_id` to support registry data sources too?

Reply:
Yeah, I think you're right. Since `get_ucr_datasource_config_by_id` can return a `StaticDataSourceConfiguration` do I need to add a check for that in `create_linked_ucr`?

Reply:
I suppose that would be a good idea. You can just use `datasource.is_static`